File: SimpleHabit/app/src/main/java/com/me/simplehabit/data/models/CategoryProgramModel.java
package com.me.simplehabit.data.models;

import com.me.simplehabit.data.vos.CategoriesProgramVO;

import java.util.List;

public interface CategoryProgramModel {

    List<CategoriesProgramVO> getCategoriesAndProgram(CategoryProgramModelDelegate delegate,boolean isForce);
    CategoriesProgramVO getCategoriesAndProgramById(String categoryId);

    interface CategoryProgramModelDelegate{

        void onCurrentProgramFetchFromNetwork(List<CategoriesProgramVO> categoriesProgramVOList);

        void onErrorOnProgramFetch(String message);
    }
}

File: SimpleHabit/app/src/main/java/com/me/simplehabit/data/models/CategoryProgramModel.kt
package com.me.simplehabit.data.models

import androidx.lifecycle.LiveData
import com.me.simplehabit.data.vos.CategoriesProgramVO

interface CategoryProgramModel {

    fun loadCategoriesAndProgram(isForce: Boolean, networkFailure: (String) -> Unit): LiveData<List<CategoriesProgramVO>>
    fun getCategoriesAndProgramById(categoryId: String): LiveData<CategoriesProgramVO>

}
